### ISSUE TITLE
VS2017 Docker-Compose Project breaks build on command line

### ISSUE THREAD

**Issue by billpratt** [dotnet/cli#6178]
## Steps to reproduce
In Visual Studio 2017: File->New Project->ASP.NET Core Web Application
Choose either Web API or Web Application
Check "Enable Docker Support"

Go to command line and run
```
dotnet restore ./[NAME].sln
dotnet build ./[NAME].sln
```

## Expected  behavior
Restore and build work when a Docker-Compose project (.dcproj) is in the solution.
Should work across all platforms that ```dotnet``` is supported

## Actual behavior
error MSB4019: The imported project "C:\Program Files\dotnet\sdk\1.0.0\Sdks\Microsoft.Docker.Sdk\Sdk\Sdk.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

```
PS C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10> dotnet --version
1.0.0

PS C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10> dotnet restore .\WebApplication10.sln
C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\docker-compose.dcproj : error MSB4019: The imported project "C:\Program Files\dotnet\sdk\1.0.0\Sdks\Microsoft.Docker.Sdk\Sdk\Sdk.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
  Restoring packages for C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\WebApplication10\WebApplication10.csproj...
  Restoring packages for C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\WebApplication10\WebApplication10.csproj...
  Restore completed in 785.82 ms for C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\WebApplication10\WebApplication10.csproj.
  Lock file has not changed. Skipping lock file write. Path: C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\WebApplication10\obj\project.assets.json
  Restore completed in 1.08 sec for C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\WebApplication10\WebApplication10.csproj.

  NuGet Config files used:
      C:\Users\foo\AppData\Roaming\NuGet\NuGet.Config
      C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

  Feeds used:
      https://api.nuget.org/v3/index.json
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
      
PS C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10> dotnet build .\WebApplication10.sln
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\docker-compose.dcproj : error MSB4019: The imported project "C:\Program Files\dotnet\sdk\1.0.0\Sdks\Microsoft.Docker.Sdk\Sdk\Sdk.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
  WebApplication10 -> C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\WebApplication10\bin\Debug\netcoreapp1.1\WebApplication10.dll

Build FAILED.

C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10\docker-compose.dcproj : error MSB4019: The imported project "C:\Program Files\dotnet\sdk\1.0.0\Sdks\Microsoft.Docker.Sdk\Sdk\Sdk.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:04.20
PS C:\Users\foo\Documents\Visual Studio 2017\Projects\WebApplication10>
```

## Environment data
`dotnet --info` output:

.NET Command Line Tools (1.0.1)

Product Information:
 Version:            1.0.1
 Commit SHA-1 hash:  005db40cd1

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.1

**Comment by livarcocc** [comment]
cc @MichaelSimons 

This is an SDK that ships only with VS. It is not supported in the CLI.

**Comment by billpratt** [comment]
Right, completely understand that. By adding the docker-compose project feature to a solution completely breaks running ```dotnet restore``` and ```dotnet build``` on a solution file. Since the dcproj is a VS2017 feature, I dont expect ```dotnet``` to build it, but perhaps ignore it. 

The microsoft/aspnetcore-build:1.0-1.1 docker image has "fixed" this by adding the SDK in the dotnet path. Howerver, I hope this is a temporary workaround.

```
root@8ca61219426a:/usr/share/dotnet# find / -xdev 2>/dev/null -name "Microsoft.Docker.Sdk"
/usr/share/dotnet/sdk/1.0.1/Sdks/Microsoft.Docker.Sdk
```

**Comment by livarcocc** [comment]
There is a PR out right now add this SDK to the CLI, which should address this issue.

https://github.com/dotnet/cli/pull/6180

**Comment by billpratt** [comment]
Great! Thanks for the quick turn around

**Comment by jgreene** [comment]
It appears this issue was closed since #6180 is supposed to resolve it.  Then #6180 was closed because it requires the Microsoft.Docker.SDK to be open sourced before it can be merged.  So how are users supposed to resolve this issue going forward?

**Comment by philcontrolf1** [comment]
@jgreene In the short term, you just need to copy the appropriate files from the VS 2017 installation (C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks\Microsoft.Docker.Sdk) to where `dotnet` expects to find them (C:\Program Files\dotnet\sdk\1.0.1\Sdks\Microsoft.Docker.Sdk on Windows, /opt/dotnet/sdk/1.0.1/Sdks/Microsoft.Docker.Sdk on *nix). Longer term hopefully the SDK will be open sourced and these files can be distributed...

**Comment by billpratt** [comment]
@livarcocc I propose that this issue be re-opened since there is no fix currently in place except for manually copying SDK files.

**Comment by jgreene** [comment]
@philcontrolf1 thank you for the solution, this worked for me

**Comment by AlexandreOuellet** [comment]
Manually copying the files also worked for me, but I also suggest to reopen this issue, as it is indeed an issue until the files are distributed with the CLI.

**Comment by philcontrolf1** [comment]
With vague apologies for the self-promotion, having put together a whole pipeline of .NET Core 1.1, Visual Studio 2017 and Docker on TeamCity, I wrote a [blog post](http://www.controlf1.co.uk/updates/may-2017/building-with-net-core-and-docker-in-teamcity/) for work on it - hopefully it will help anyone else trying to do this.

**Comment by khayes** [comment]
I also ran into this today. Copying the files is non-ideal as we run `dotnet restore` on a dynamic build system and having to do this on every build agent is awkward.

**Comment by philcontrolf1** [comment]
@khayes Our build agent bootstrapper copies the files on - see my blog post for details.

**Comment by opless** [comment]
In the community edition the sdk is: 
"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\Sdks\Microsoft.Docker.Sdk"

... and for 2.0.0-preview1-005977 on OSX you need to put the files here: 
/usr/local/share/dotnet/sdk/2.0.0-preview1-005977/Sdks/

**Comment by whyleee** [comment]
I have 2.0.0-preview2-006497 version installed on Mac and copying Microsoft.Docker.Sdk folder doesn't help there. I get another error with default .NET Core 2.0 api project and Docker support:
```
$ dotnet run
/usr/local/share/dotnet/sdk/2.0.0-preview2-006497/Sdks/Microsoft.Docker.Sdk/build/Microsoft.VisualStudio.Docker.Compose.targets(80,45): error MSB4022: The result "" of evaluating the value "$(DockerBuildTasksAssembly)" of the "AssemblyFile" attribute in element <UsingTask> is not valid. [/Users/admin/git/test/docker-compose.dcproj]

The build failed. Please fix the build errors and run again.
```

**Comment by rainersigwald** [comment]
Since these extensions are installed to VS's copy of msbuild but not `dotnet`'s, can you run the build with `msbuild.exe` instead of `dotnet build`?

**Comment by philcontrolf1** [comment]
@rainersigwald Not on my Linux CI build agents I can't :-)

**Comment by srivatsn** [comment]
Reopening this issue since it hasn't been resolved yet. I think not being able to build a solution that contains a dcproj without hacks is a significant blocker.

@dazhao-msft - any updates on https://github.com/dotnet/cli/pull/6180

**Comment by billpratt** [comment]
@srivatsn thank you for re-opening

**Comment by dmitry606** [comment]
The fix (copying VS Docker SDK to dotnet sdk) does not seem to work for 2.0.0-preview2. I get this error:
C:\Program Files\dotnet\sdk\2.0.0-preview2-006497\Sdks\Microsoft.Docker.Sdk\build\Microsoft.VisualStudio.Docker.Compose.targets(80,45): error MSB4022: The result "" of evaluating the value "$(DockerBuildTasksAssembly)" of the "AssemblyFile" attribute in element <UsingTask> is not valid. [C:\repos\WebApplication1\docker-compose.dcproj]

Any solutions for this?

**Comment by philcontrolf1** [comment]
@dmitry606 Seems to work for me:

    PS D:\Source\HelloWorld> dir

    Directory: D:\Source\HelloWorld

    Mode                LastWriteTime         Length Name
    ----                -------------         ------ ----
    d-----       01/08/2017     09:15                bin
    d-----       01/08/2017     09:15                HelloWorld
    d-----       01/08/2017     09:15                obj
    -a----       01/08/2017     09:15            271 docker-compose.ci.build.yml
    -a----       01/08/2017     09:15            723 docker-compose.dcproj
    -a----       01/08/2017     09:15             14 docker-compose.override.yml
    -a----       01/08/2017     09:15            370 docker-compose.vs.debug.yml
    -a----       01/08/2017     09:15            264 docker-compose.vs.release.yml
    -a----       01/08/2017     09:15            136 docker-compose.yml
    -a----       01/08/2017     09:15           1479 HelloWorld.sln


    PS D:\Source\HelloWorld> dotnet --version
    2.0.0-preview2-006497
    PS D:\Source\HelloWorld> dotnet build .\HelloWorld.sln
    Microsoft (R) Build Engine version 15.3.388.41745 for .NET Core
    Copyright (C) Microsoft Corporation. All rights reserved.

      HelloWorld -> D:\Source\HelloWorld\HelloWorld\bin\Debug\netcoreapp1.1\HelloWorld.dll

    Build succeeded.
        0 Warning(s)
        0 Error(s)

    Time Elapsed 00:00:01.97
    PS D:\Source\HelloWorld> dotnet run --project .\HelloWorld\HelloWorld.csproj
    Hello World!

Have you got exactly the same SDK files in exactly the same location?

    $ sha1sum /cygdrive/c/Program\ Files/dotnet/sdk/2.0.0-preview2-006497/Sdks/Microsoft.Docker.Sdk/Sdk/*
    a30e05d5e5d250884a2e5a29f4730f95e0fb6f85 */cygdrive/c/Program Files/dotnet/sdk/2.0.0-preview2-006497/Sdks/Microsoft.Docker.Sdk/Sdk/Sdk.props
    cea93148ba2a590fbda25bfba0ed561f510e6c4a */cygdrive/c/Program Files/dotnet/sdk/2.0.0-preview2-006497/Sdks/Microsoft.Docker.Sdk/Sdk/Sdk.targets

(`/cygdrive/c` is just `c:\` in case you're not a Cygwin person)

**Comment by dmitry606** [comment]
@philcontrolf1 Yes, it's exactly the same location

`c:\repo\HelloWorld>cd C:\Program Files\dotnet\sdk\2.0.0-preview2-006497\Sdks\Microsoft.Docker.Sdk\Sdk

C:\Program Files\dotnet\sdk\2.0.0-preview2-006497\Sdks\Microsoft.Docker.Sdk\Sdk>dir

 Directory of C:\Program Files\dotnet\sdk\2.0.0-preview2-006497\Sdks\Microsoft.Docker.Sdk\Sdk

07/31/2017  03:59 PM    <DIR>          .
07/31/2017  03:59 PM    <DIR>          ..
07/28/2017  06:50 PM             1,927 Sdk.props
07/28/2017  06:50 PM             1,220 Sdk.targets
               2 File(s)          3,147 bytes
`

**Comment by philcontrolf1** [comment]
@dmitry606 You've definitely got different files though - my `Sdk.props` is 1750 bytes and my `Sdk.targets` is 1264 bytes. Both copied from `C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks\Microsoft.Docker.Sdk\Sdk` with Visual Studio 15.2.26430.16, although I don't think they've changed in a while.

**Comment by dmitry606** [comment]
@philcontrolf1 Yep.. looks like that. I copied the SDK from VS 2017 Preview 2.

**Comment by dazhao-msft** [comment]
@dmitry606 It looks like that you copied the Microsoft.Docker.Sdk folder from VS 15.3 Preview, under which there are 3 folders: build, Sdk, tools. Please delete build and tools and keep Sdk only.

**Comment by dmitry606** [comment]
@dazhao-msft It worked! Thanks!!

**Comment by milla** [comment]
now Linux .net core 2.0 sdk doesn't support docker? I got 
`docker-compose.dcproj : error MSB4236: The SDK 'Microsoft.Docker.Sdk' specified could not be found.
`

**Comment by philcontrolf1** [comment]
@milla Have you copied the appropriate SDK as listed in this thread? If so, *exactly* which files did you copy, where did you copy them from, and where did you copy them to? A number of us have this working under Preview 2, so it's definitely not completely broken.

**Comment by milla** [comment]
@philcontrolf1 thanks for checking. It's confirmed that this is a mistake, I built many vsts build agents, some of them don't have docker installed. and the agent I used and occurred this problem just doesn't have. Now everything is working perfectly.

**Comment by liri2006** [comment]
Any ETA on fix?

**Comment by mmisztal1980** [comment]
This is still broken after .NET Core 2.0 release. Can anyone comment on the state of affairs?

I've tried to do a `dotnet build ./MySolution.sln` on a solution which contained a project with enabled Docker (linux) support.

**Comment by philcontrolf1** [comment]
@mmisztal1980 You still need to copy the files in as detailed here, because Microsoft still haven't open-sourced them. @srivatsn, @dazhao-msft: it would be *really* good if we could have a progress update here, even if it's just "we're working on it". As you can see, this is hitting a lot of people.

**Comment by dasMulli** [comment]
While waiting for it to be open sourced was a good idea when the expectation was that it would be open sourced in a few weeks time, it is now becoming increasingly cumbersome.
Even mono added the stub SDK to its msbuild distribution https://github.com/mono/msbuild/tree/xplat-master/sdks/Microsoft.Docker.Sdk

**Comment by philcontrolf1** [comment]
@dasMulli Interesting, they *are* the official files. Not quite sure how they're ending up in the MIT licensed Mono with a line in them saying "Copyright (C) Microsoft Corporation. All rights reserved." though...

**Comment by billpratt** [comment]
@livarcocc Can we please have an update on this issue?

**Comment by davidmceleney** [comment]
A workaround to this - if you're using teamcity, change your build script to execute the commands in your docker-compose file... 

Replace:
`docker-compose docker-compose docker-compose.ci.build.yml up`

With:
`dotnet restore mysolution.sln
dotnet publish mysolution.sln -c Release -o ./obj/Docker/publish`

removes the dependency on Microsoft.Docker.Sdk... 

hope it helps someone!

**Comment by matthewDDennis** [comment]
I found a work around.  Set the build dependencies on the docker-compose project to include ALL other projects.  This will allow the other projects to build/publish before dotnet attempts to build/publish the docker project.  You still get an error at the end, but it does not abort the build/publish until it doesn't matter.

**Comment by philcontrolf1** [comment]
@FelschR No, it's never worked.

**Comment by FelschR** [comment]
@philcontrolf1 Interesting... I'm using VSTS and it definetely works with .NET Core 1.1.
Maybe the previous `aspnetcore-build` docker image shipped with the Microsoft.Docker.Sdk?

**Comment by philcontrolf1** [comment]
@FelschR Maybe. It certainly doesn't work if you build your images from scratch, just installing the .NET SDK on top of an empty Linux image.

**Comment by billpratt** [comment]
@FelschR 

> Maybe the previous aspnetcore-build docker image shipped with the Microsoft.Docker.Sdk?

The microsoft/aspnetcore-build:1.0-1.1 docker image "fixed" this by adding the SDK in the dotnet path.

**Comment by billpratt** [comment]
@livarcocc @dazhao-msft @eerhardt Can we please have an update on this? Even a "we are working on it.." would be helpful. More and more people are coming to this issue as well as the [closed PR](6180) that would have fixed it.

**Comment by FelschR** [comment]
For all those trying to get the docker image (`microsoft/aspnetcore-build`) working see:
https://github.com/aspnet/aspnet-docker/issues/299

Apparently there is also a `microsoft/aspnetcore-build:1.0-2.0` image that comes with the Microsoft.Docker.Sdk.

Finally my build is working again thanks to @natemcmaster who provided that solution.

**Comment by badavid85** [comment]
Just to summarize the steps I had to perform to make it work on Windows and Linux (Ubuntu):

1. Copy the _Microsoft.Docker.Sdk_ folder from _C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks_ (**Only** the Sdk subfolder. **Do not** copy _build_ and _tools_ subfolders).
2. Then paste it on _C:\Program Files\dotnet\sdk\2.0.0\Sdks_ on Windows and _/usr/share/dotnet/sdk/2.0.0./Sdks_ on Linux (Ubuntu)
Then `dotnet build SolutionName.sln` will work fine.

These steps will fix both errors:
- error MSB4236: The SDK 'Microsoft.Docker.Sdk' specified could not be found.
- error MSB4022: The result "" of evaluating the value "$(DockerBuildTasksAssembly)" of the "AssemblyFile" attribute in element is not valid.

**Comment by mkoertgen** [comment]
Ok, the suggested workaround fixes `dotnet restore` and `dotnet build`. 

How about `dotnet run`?

```bash
Unable to run your project.
Please ensure you have a runnable project type and ensure 'dotnet run' supports this project.
A runnable project should target a runnable TFM (for instance, netcoreapp2.0) and have OutputType 'Exe'.
The current OutputType is 'DockerCompose'.
```

Seems that the output type is not supported. Any idea on how to work around this, too?

**Comment by andrewsav-datacom** [comment]
@mkoertgen The way I read this thread is that the visual studio functionality is not intended to work with dotnet core cli. The workaround resolves the inability to build the solution by skipping the docker compose project in it. Of course you cannot *run* it after this because it simply has not been built.

**Comment by mlowijs** [comment]
As another workaround, one could run `dotnet sln MySolution.sln remove docker-compose.dcproj` to remove the project reference.

This is what I do in my VSTS build definition.

**Comment by Vlaaaaaaad** [comment]
Will this ever be fixed?

**Comment by sheigel** [comment]
After so many years of generating frustration it seems that Visual Studio still takes priority to having a working/proper CLI.
It seems that having a strong CLI that always works is much more useful than having a slow running IDE that should not be installed on the CI machine. But for the sake of having this single IDE more click-friendly you are willing to break something so foundational as a platform independence.

**Comment by billpratt** [comment]
@DamianEdwards @shanselman Sorry to include you two on this issue but we haven't received feedback on this issue for some time now and we are looking for help. More and more people are coming to this issue as well as the [closed PR](6180) that would have fixed it.

**Comment by tyronegroves** [comment]
So no fix?

**Comment by shanselman** [comment]
ping @richlander ?

**Comment by zunair-ch** [comment]
I open Docker CLI as an administrator and just move to the project directory and run following set of commands and it works for me. Also make sure kitematics is running if you are on windows 8.
1) dotnet restore
2) dotnet build
3) dotnet run

**Comment by mmisztal1980** [comment]
Is this going to get fixed with .net core 2.1?

**Comment by zunair-ch** [comment]
Yes.For me, it's working for .net core 2.0 and 2.1

**Comment by philcontrolf1** [comment]
@zunair-ch Yes, it will work in the *project* directory because then all you're building is the one project. What happens if you build the *solution* - e.g. `dotnet build ./MySolution.sln` in the solution directory?

**Comment by philcontrolf1** [comment]
For all the watchers here: #8416 should fix this one. I've just bumped it to see if there's anything blocking it.

**Comment by billpratt** [comment]
This a good to hear. Thanks for fixing!

**Comment by zunair-ch** [comment]
Thanks all.

**Comment by TomaszG** [comment]
I'm currently facing the same problem - solution with docker-compose.dcproj (added by VS after adding enabling docker support) cannot be built or restored with dotnet CLI. 
`C:\Program Files\dotnet\sdk\2.1.301\NuGet.targets(114,5): error : Invalid restore input. Invalid target framework 'unsupported'. Input files: C:\project\src\docker-compose.dcproj. [C:\project\src\something.sln]`

Looking at the comments above I think it should be working, however it doesn't (at least for me). Am I missing something?

Below is **dotnet --info** output:
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.301
 Commit:    59524873d6

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.301\

Host (useful for support):
  Version: 2.1.1
  Commit:  6985b9f684

.NET Core SDKs installed:
  1.0.0 [C:\Program Files\dotnet\sdk]
  2.1.4 [C:\Program Files\dotnet\sdk]
  2.1.101 [C:\Program Files\dotnet\sdk]
  2.1.102 [C:\Program Files\dotnet\sdk]
  2.1.103 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.201 [C:\Program Files\dotnet\sdk]
  2.1.301 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

```

**Comment by joergjo** [comment]
The latest Docker support for ASP.NET Core projects doesn't use `.dcproj` anymore. So if you recreate your project structure with the latest tooling, you won't run into this issue anymore.

**Comment by gitfool** [comment]
@joergjo I tried; that only works if a single `Dockerfile` suffices. If you need to use `docker-compose`, after creating a new project, right-click `Add Orchestrator Support` and it adds a `dcproj` file to the `sln` and refers to it in the WebApplication `csproj`:

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
    <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
    <DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>
    <UserSecretsId>f3532c1f-c8ec-4713-8e86-0fc6ebc385c8</UserSecretsId>
    <DockerComposeProjectPath>..\docker-compose.dcproj</DockerComposeProjectPath>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" />
    <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.2.0" PrivateAssets="All" />
    <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.0.2105168" />
  </ItemGroup>
</Project>
```
... and the command-line build fails again:
```
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Program Files\dotnet\sdk\2.2.101\NuGet.targets(114,5): error : Invalid restore input. Invalid target framework 'unsupported'. Input files: D:\Devel\Repos\WebApplication1\docker-compose.dcproj. [D:\Devel\Repos\WebApplication1\WebApplication1.sln]

Build FAILED.
```
There is a newer version available on NuGet, compared to what was generated by the right-click experience, of [Microsoft.VisualStudio.Azure.Containers.Tools.Targets](https://www.nuget.org/packages/Microsoft.VisualStudio.Azure.Containers.Tools.Targets), but that doesn't help.

I don't know the details, but it seems to me that we're _still_ blocked on the `Microsoft.Docker.Sdk` being open sourced before we can use the `dotnet` CLI to build these solutions on Linux.

**Comment by joergjo** [comment]
I hadn't even noticed that feature! 

I always use the "basic" Docker support (i.e. no `.dcproj`), and add `docker-compose` files manually as required. 

If I want to debug a containerized app with its dependencies running in Docker as well (which rarely happens), I start all dependencies using `docker-compose` before debugging the actual app, and expose their ports on the host. In this setup, the application can resolve its dependencies using `host.docker.internal` as host name. It's requires a few more manual steps and works only for a debugging a single container, but it saves me all the `.dcproj` headache you've mentioned.

**Comment by vchirikov** [comment]
`3.0.100-preview8-013656` still broken.
@livarcocc can you reopen this?

**Comment by livarcocc** [comment]
@vchirikov how is it broken for you?

**Comment by vchirikov** [comment]
@livarcocc you can read about it here https://github.com/microsoft/DockerTools/issues/209

**Comment by livarcocc** [comment]
Looking at the issue you referenced, the CLI team will not enable building this dcprojs in the command line. Though, there is now an SDK that should at least skip doing anything for it.

What seems like a bug to me is the failure to use dotnet sln add for dcprojs. If that's the case, I would suggest filling a brand new issue just for dotnet sln add.

**Comment by LagManos** [comment]
@livarcocc Which SDK should skip dcprojs while building? I've just tested with 3.1.100-preview1-014459 and I'm unable to run dotnet build or dotnet test on solution containing dcproj file. The error is:  C:\Program Files\dotnet\sdk\3.1.100-preview1-014459\NuGet.targets(123,5): error : Invalid restore input. Invalid target framework 'unsupported'.

**Comment by livarcocc** [comment]
Can you please file a separate and new issue for this? So that we can start fresh?!

@dsplaisted wonder if this is due to the framework composition work we did. Could you take a look?

**Comment by LagManos** [comment]
Unfortunately I can't reproduce it with solution created from scratch with VS2017, VS2019 and different project types. Issue occurs only in my old and big repository, which I can't share. 

Maybe there is a corner case that I have run into, but I can't really tell exactly what it is.

**Comment by vchirikov** [comment]
@livarcocc repro here
https://github.com/microsoft/DockerTools/issues/209#issuecomment-551064216
